components/camera/camera.go

```diff
@@ -192,6 +192,14 @@ type videoSource struct {
 	streamType   StreamType
 }
 
+// SourceFromCamera returns a gostream.VideoSource from a camera.Camera if possible, else nil.
+func SourceFromCamera(cam Camera) (gostream.VideoSource, error) {
+	if asSrc, ok := cam.(*videoSource); ok {
+		return asSrc.videoSource, nil
```

Comment:
camera's already fulfill the VideoSource interface too, you could also just use cam.(gostream.VideoSource) directly. Unless you want to get rid of all the extra camera methods on purpose

Reply:
So it compiles but it does not work and initially I was confused as to why it wouldn't. I printed out the type info to help my understanding.
```
fmt.Printf("cam: %T\n", cam)
fmt.Printf("asSrc: %T\n", asSrc)
fmt.Printf("asSrc.videoSource: %T\n", asSrc.videoSource)
...
cam: *camera.videoSource
asSrc: *camera.videoSource
asSrc.videoSource: *gostream.mediaSource[image.Image,github.com/pion/mediadevices/pkg/prop.Video]
```

What we ultimately need is the last type, a `gostream.mediaSource[image.Image, prop.Video]` which contains the `driver` we'll eventually query to retrieve its properties. We can't access any of the internals of camera.Camera (it's just an interface) so the chain of casting calls is used to cast that interface into types we know the internals of.

Trying to do all of this in one function wouldn't work either since that would require `gostream` to know about the private struct `camera.videoSource`.